### PR TITLE
Use std::array instead of GG::Clr in common and server code

### PR DESCRIPTION
**Pull request by o01eg**
Removes usage of GG in the non-client code.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -684,18 +684,19 @@ endif()
...
 ##
 
-add_subdirectory(client)
-add_subdirectory(combat)
-add_subdirectory(Empire)
-add_subdirectory(network)
-add_subdirectory(parse)
-add_subdirectory(python)
-add_subdirectory(server)
+add_subdirectory(src/common/client)
```
Comment (geoffthemedio) [inline CMakeLists.txt]: can you separate the source layout changes?

Comment (o01eg) [reply CMakeLists.txt]: It's here #3137

Comment (o01eg) [reply CMakeLists.txt]: Removed sources layout changes.

File: server/UniverseGenerator.cpp [hunk 2]
```diff
@@ -875,18 +875,20 @@ void InitEmpires(const std::map<int, PlayerSetupData>& player_setup_data) {
...
             } else {
                 // as a last resort, make up a colour
-                empire_colour = GG::FloatClr(static_cast<float>(RandZeroToOne()),
-                                             static_cast<float>(RandZeroToOne()),
-                                             static_cast<float>(RandZeroToOne()),
-                                             1.0f);
+                empire_colour = {RandInt(0, 255),
+                                 RandInt(0, 255),
+                                 RandInt(0, 255),
+                                 255};
```
Comment (geoffthemedio) [inline server/UniverseGenerator.cpp:891]: https://ci.appveyor.com/project/freeorion/freeorion/builds/35875836/job/78bhaug75kmcby0b#L275
````
error C2397: conversion from 'int' to '_Ty' requires a narrowing conversion [C:\projects\freeorion\msvc2017\FreeOrionD\FreeOrionD.vcxproj]
          with
          [
              _Ty=unsigned char
          ]
````

Comment (o01eg) [reply server/UniverseGenerator.cpp:891]: Should be fixed with `static_cast`.

**Comment by geoffthemedio** [comment]
Can this load saves that serialized `GG::Clr`?

**Comment by o01eg** [comment]
@geoffthemedio No, that why I asked about breaking compatibility on forum.

**Comment by geoffthemedio** [comment]
That needs to be fixed for the medium term. After the next version, the compatibility can be dropped, but it should load old saves for now. Workaround could involve a local `Clr` definition in the serialization code, if that helps....

**Comment by o01eg** [comment]
@geoffthemedio Restored compatibility.

File: client/human/HumanClientApp.cpp [hunk 3]
```diff
@@ -569,12 +569,12 @@ void HumanClientApp::NewSinglePlayerGame(bool quickstart) {
...
     // DB stores index into array of available colours, so need to get that array to look up value of index.
     // if stored value is invalid, use a default colour
-    const std::vector<GG::Clr>& empire_colours = EmpireColors();
+    const std::vector<EmpireColor>& empire_colours = EmpireColors();
     int colour_index = GetOptionsDB().Get<int>("setup.empire.color.index");
     if (colour_index >= 0 && colour_index < static_cast<int>(empire_colours.size()))
         human_player_setup_data.empire_color = empire_colours[colour_index];
     else
-        human_player_setup_data.empire_color = GG::CLR_GREEN;
+        human_player_setup_data.empire_color = {GG::CLR_GREEN.r, GG::CLR_GREEN.g, GG::CLR_GREEN.b, GG::CLR_GREEN.a};
```
Comment (geoffthemedio) [inline client/human/HumanClientApp.cpp:577]: should probably add a conversion operator later

**Pull request merged**